### ISSUE TITLE
Steps to unblock the npm release

### ISSUE THREAD

**Issue by ad1992** [excalidraw/excalidraw#5960]
After the Editor redesign, there has been a huge refactor the way the components are designed hence we want to fix the issues related to the refactor in the npm package and post that we will unblock the package release.

Below are the steps needed to achieve the same

- [x] The `renderFooter` prop seems to be broken CSS wise, 
- [ ] The Excalidraw App code is now getting clubbed in npm package. This is I think due to import of `languageList` from excalidraw app 
- [ ] I have a WIP PR for rendering custom footer as a component, will try to merge that as well https://github.com/excalidraw/excalidraw/pull/5436/
- [ ] The `language` dropdown was earlier part of `renderFooter` but now a dropdown is introduced and the language dropdown is part of the dropdown Create an API to so that host could add additional items in the dropdown and render it their way. With this change rendering, the dropdown will be completely the responsibility of host

Proposed API

```js
<HamburgerMenu>
<CanvasOptions/>
<HamburgerMenuItem onClick={onLangChange}>
<LangDropDown/>
</HamburgerMenuItem>
<SocialOptions/>
</HamburgerMenu>
```

- [ ] Should welcome screen be made optional ? May be not as its helps to onboard new users

What else ?

cc / @dwelle

**Comment by ad1992** [comment]
@dwelle I think we can unblock the `next ` tagged release once the footer CSS is fixed ?
Rest all changes will be done before the stable release of the next version.

**Comment by dwelle** [comment]
> @dwelle I think we can unblock the `next ` tagged release once the footer CSS and language import is fixed? Rest all changes will be done before the stable release of the next version.

I don't think releasing `@next` before we allow customization of the hamburger menu and the welcome screen will be of much help to host apps.

**Comment by ad1992** [comment]
> I don't think releasing @next before we allow customization of the hamburger menu and the welcome screen will be of much help to host apps.

True but since its `@next` so it's expected to change and not everyone is going to customize the hamburger / welcome screen hence I still think it is ok to unblock the next releases so users can start using it

**Comment by ad1992** [comment]
<img width="860" alt="Screenshot 2022-12-09 at 4 37 29 PM" src="https://user-images.githubusercontent.com/11256141/206660100-6875ab4d-62a6-43f7-80fe-e9f2d2467522.png">

@dwelle ^

**Comment by dwelle** [comment]
You're highlighting Language.

Doesn't moving canvas background and dark mode above make more sense?

![image](https://user-images.githubusercontent.com/5153846/206676393-c848bebf-8186-4c5e-a146-44f6ee5c4e9b.png)

Problem is with Live Collab, which is also custom.

**Comment by ad1992** [comment]
> You're highlighting Language.
> 
> Doesn't moving canvas background and dark mode above make more sense?
> 
> ![image](https://user-images.githubusercontent.com/5153846/206676674-f3f1b332-dd1d-494a-b40d-4ccdd24dd4d2.png)
> 
> Problem is with Live Collab, which is also custom.

In that case, probably we will need to allow appending custom items  in both sections but "Live collab" we will not want to move in the end so then this would end up becoming a full-fledged host API where the host is responsible for rendering the menu completely

**Comment by dwelle** [comment]
If we ignore Live collaboration for now (we still allow to disable it from the UIOptions), then we can shuffle the rest so that we only need append on end:

![image](https://user-images.githubusercontent.com/5153846/208869396-e163d56f-be92-46d6-b0ba-58b5d18ba74c.png)

**Comment by dwelle** [comment]
I've opened a separate issue to discuss HamburgerMenu https://github.com/excalidraw/excalidraw/issues/6023

**Comment by ad1992** [comment]
I think "WelcomeScreen" and decoupling "Collab" related things we should do definitely do but not a blocker for the package release. 
We can release the package once menu component API is merged so users can start migrating to new version and in future version we can support more component level APIs+customisation.

**Comment by ad1992** [comment]
The removal of the clubbing npm package has been already taken care in https://github.com/excalidraw/excalidraw/pull/6034 :)

**Comment by dwelle** [comment]
> I think "WelcomeScreen" and decoupling "Collab" related things we should do definitely do but not a blocker for the package release. We can release the package once menu component API is merged and the clubbing of excal app is removed so users can start migrating to new version and in future version we can support more component level APIs+customisation.

Missed your comment yesterday. As discussed on call, we def need to bundle the WelcomeScreen in the release due to being tied to excalidraw.com.

We also need this fix for the Menu and related rewrite https://github.com/excalidraw/excalidraw/pull/6042 (but it's tiny).

And I'd argue we should bundle the `onCollabButtonClick` rewrite as well because we'd be shipping a half-way rewrite (introducing `MainMenu.DefaultItems.LiveCollaboration`).

**Comment by ad1992** [comment]
The package `@excalidraw/excalidraw@next` has been unblocked so users can try out the `next` version. 
We will release the stable version very very soon!

**Comment by ad1992** [comment]
closing this as the package was released [@excalidraw/excalidraw](https://www.npmjs.com/package/@excalidraw/excalidraw)
